File: polynomial regression.R
dataset= read.csv('Position_Salaries.csv')
dataset=dataset[2:3]

#library(caTools)
#set.seed(123)
#split=sample.split(dataset$Profit,SplitRatio=0.8)
#training_set=subset(dataset,split==TRUE)
#test_set=subset(dataset,split==FALSE)


#fitting linear regression to the dataset

lin_reg=lm(formula=Salary~.,
           data = dataset)
#fitting polynomial regression to the dataset
dataset$Level2=dataset$Level^2
dataset$Level3=dataset$Level^3
dataset$Level4=dataset$Level^4
poly_reg=lm(formula=Salary~.,
            data=dataset)
#visualising the linear reg model
ggplot()+
  geom_point(aes(x=dataset$Level,y=dataset$Salary),color='red')+
  geom_line(aes(x=dataset$Level,y=predict(lin_reg,newdata = dataset)),color='blue')+
  ggtitle('truth or bluff(linear regression)')+
  xlab('Level')+
  ylab('salary')
#visualising the polynomial reg modellibrary(ggplot2)
ggplot()+
  geom_point(aes(x=dataset$Level,y=dataset$Salary),color='red')+
  geom_line(aes(x=dataset$Level,y=predict(poly_reg,newdata = dataset)),color='blue')+
  ggtitle('truth or bluff(linear regression)')+
  xlab('Level')+
  ylab('salary')

#predicting the salary with linear reg model


y_pred=predict(lin_reg,data.frame(Level=6.5))

#prediction with polynmial reg model
y_pred=predict(poly_reg,data.frame(Level=6.5,Level2=6.5^2,Level3=6.5^3,Level4=6.5^4))

File: polynomial regression.py
# -*- coding: utf-8 -*-
"""
Created on Tue Mar  7 13:40:42 2017

@author: Habiba
"""

import numpy as np
import matplotlib.pyplot as plt
import pandas as pd


dataset=pd.read_csv('Position_Salaries.csv')
X=dataset.iloc[:,1:2].values
y=dataset.iloc[:,2].values              

"""#splitting dataset into training and data set
from sklearn.cross_validation import train_test_split
X_train,X_test,y_train,y_test=train_test_split(X,y,test_size=0.2,random_state=0)"""

#feature scaling
"""from sklearn.preprocessing import StandardScaler
sc_X=StandardScaler()
X_train=sc_X.fit_transform(X_train)
X_test=sc_X.transform(X_test)"""
#linear regression model
from sklearn.linear_model import LinearRegression
lin_reg=LinearRegression()
lin_reg.fit(X,y)

#fitting polynomial regression
from sklearn.preprocessing import PolynomialFeatures
poly_reg=PolynomialFeatures(degree=4)
X_poly=poly_reg.fit_transform(X)

lin_reg_2=LinearRegression()
lin_reg_2.fit(X_poly,y)

#visualize the results
plt.scatter(X,y,color='red')
plt.plot(X,lin_reg.predict(X),color='blue')
plt.title('regression graph(linear)')
plt.xlabel('position level')
plt.ylabel('salary')
plt.show()
#visualise the polynomial regression
X_grid=np.arange(min(X),max(X),0.1)
X_grid=X_grid.reshape(len(X_grid),1)
plt.scatter(X,y,color='red')
plt.plot(X_grid,lin_reg_2.predict(poly_reg.fit_transform(X_grid)),color='blue')
plt.title('regression graph(polynomial)')
plt.xlabel('position level')
plt.ylabel('salary')
plt.show()

#predicting the result with linear regresion
lin_reg.predict(6.5)

#predicting the result with polynomial regresion
lin_reg_2.predict(poly_reg.fit_transform(6.5))
